articles/node_lifecycle.md

```diff
@@ -0,0 +1,287 @@
+way defined in the life cycle state machine is malformed.
+
+These services may be provided via attributes and method calls (for local
+management) or via ROS messages and services (for remote management).
+In the case of providing a ROS middleware interface, specific topics must be
+used, and they should be placed in a suitable namespace.
+
+### Initialisation
+
+The following service should be called to initialise the node. This service must
+be named `initialise`.
+
+{% raw %}
+    ---
+    bool success
+    string result_msg
+{% endraw %}
+
+
+### Activation
+
+The following service should be called to activate the node. This service must
+be named `activate`.
+
+{% raw %}
+    ---
+    bool success
+    string result_msg
+{% endraw %}
+
+
+### Deactivation
+
+The following service should be called to deactivate the node. This service
+must be named `deactivate`.
+
+{% raw %}
+    ---
+    bool success
+    string result_msg
+{% endraw %}
+
+
+### Clearing the error state
+
+The following service should be called to clear the node's fatal error status.
+This is done when the node is to be re-activated in the hope that the error has
+gone away.
+
+{% raw %}
+    ---
+    bool success
+{% endraw %}
+
+### State reporting
+
+The following service should be called to query the current state of the node.
+The service must be named `get_state`.
+
+{% raw %}
+    ---
+    int8 created=0
+    int8 inactive=1
+    int8 active=2
+    int8 error=3
+    int8 destroyed=4
+
+    int8 state
+{% endraw %}
+
+The value of `state` must be one of the constants defined in the service
+definition.
+
+
+## Composite nodes
+
+A composite node is a system-level concept of one node being made up of one or
+more other nodes as its implementation, rather than direct implementation using
+source code.
+
+It is conceivable that all child nodes in a composite node will be registered
+to a single Executor that is managed by the composite node. Such a design
+allows for grouping nodes into a single process, such as for control loops.
+(See the Executor examples on GitHub for an example of this.)
+
+A composite node should enforce its own life cycle state on all its children
+according to the following mapping:
+
+<table>
+    <tr><td>Composite state</td><td>Child state</td></tr>
+    <tr><td>Created</td><td>Created</td></tr>
+    <tr><td>Inactive</td><td>Inactive</td></tr>
+    <tr><td>Active</td><td>Active</td></tr>
+    <tr><td>Fatal error</td><td>Inactive</td></tr>
+    <tr><td>Destroyed</td><td>Destroyed</td></tr>
+</table>
+
+If the composite node suffers a fatal error, then its children must be
+deactivated.
+
+If one of the child nodes has suffered a fatal error, then the composite must
+be shifted to the `fatal error` state and all other child nodes shifted to the
+`inactive state`.
+
+## Composite node services
+
+Composite nodes should provide the following services to allow adding and
```

Comment:
This also means that the parent must be in `inactive state`, as per the above table.